tap_salesforce/__init__.py

```diff
@@ -268,47 +338,32 @@ def do_discover(sf):
     entries = []
 
     # Check if the user has BULK API enabled
-    if sf.api_type == 'BULK' and not Bulk(sf).has_permissions():
-        raise TapSalesforceBulkAPIDisabledException('This client does not have Bulk API permissions, received "API_DISABLED_FOR_ORG" error code')
-
-    for sobject_name in sorted(objects_to_discover):
-
-        # Skip blacklisted SF objects depending on the api_type in use
-        # ChangeEvent objects are not queryable via Bulk or REST (undocumented)
-        if (sobject_name in sf.get_blacklisted_objects() and sobject_name not in ACTIVITY_STREAMS) \
-           or sobject_name.endswith("ChangeEvent"):
-            continue
-
-        sobject_description = sf.describe(sobject_name)
-
-        if sobject_description is None:
-            continue
-
-        # Cache customSetting and Tag objects to check for blacklisting after
-        # all objects have been described
-        if sobject_description.get("customSetting"):
-            sf_custom_setting_objects.append(sobject_name)
-        elif sobject_name.endswith("__Tag"):
-            relationship_field = next(
-                (f for f in sobject_description["fields"] if f.get("relationshipName") == "Item"),
-                None)
-            if relationship_field:
-                # Map {"Object":"Object__Tag"}
-                object_to_tag_references[relationship_field["referenceTo"]
-                                         [0]] = sobject_name
-
-        fields = sobject_description['fields']
-        replication_key = get_replication_key(sobject_name, fields)
-
-        # Salesforce Objects are skipped when they do not have an Id field
-        if not [f["name"] for f in fields if f["name"]=="Id"]:
-            LOGGER.info(
-                "Skipping Salesforce Object %s, as it has no Id field",
-                sobject_name)
-            continue
+    if sf.api_type == "BULK" and not Bulk(sf).has_permissions():
+        raise TapSalesforceBulkAPIDisabledException(
+            'This client does not have Bulk API permissions, received "API_DISABLED_FOR_ORG" error code'
+        )
 
-        entry = generate_schema(fields, sf, sobject_name, replication_key)
-        entries.append(entry)
+    objects_list = sorted(objects_to_discover)
+    start_counter = 0
+    concurrency_limit = 25
+
+    while start_counter < len(objects_list):
+        end_counter = start_counter + concurrency_limit
+        if end_counter >= len(objects_list):
+            end_counter = len(objects_list)
+
+        chunk = objects_list[start_counter:end_counter]
+        chunk_args = [
+            [
+                sf,
+                sobject_name,
+                entries,
+                sf_custom_setting_objects,
+                object_to_tag_references,
+            ]
+            for sobject_name in chunk]
+        run_concurrently(discover_stream, chunk_args)
+        start_counter = end_counter
```

Comment:
```suggestion
    objects_list = sorted(objects_to_discover)
    concurrency_limit = 25

    for start_counter in range(0, len(objects_list), concurrency_limit):
        chunk = objects_list[start_counter:start_counter + concurrency_limit]
        chunk_args = [
            (sf, sobject_name, entries, sf_custom_setting_objects, object_to_tag_references)
            for sobject_name in chunk
        ]
    
        run_concurrently(discover_stream, chunk_args)
```
Seems a little bit clearer like that